proposals/NNNN-init-accessors.md

```diff
@@ -0,0 +1,362 @@
+
+Property wrappers support bespoke definite initialization that allows initializing the backing property wrapper storage via the computed property, always re-writing initialization-via-wrapped-property in the form `self.value = value` to initialization of the backing storage in the form of `_value = Wrapper(wrappedValue: value)`:
+
+```swift
+@propertyWrapper
+struct Wrapper<T> {
+  var wrappedValue: T
+}
+
+struct S {
+  @Wrapper var value: Int
+
+  init(value: Int) {
+    self.value = value  // Re-written to self._x = Wrapper(wrappedValue: value)
+  }
+
+  init(other: Int) {
+    self._value = Wrapper(wrappedValue: other) // Okay, initializes storage '_x' directly
+  }
+}
+```
+
+The ad-hoc nature of property wrapper initializers mixed with an exact definite initialization pattern prevent property wrappers with additional arguments from being initialized out-of-line. Furthermore, property-wrapper-like macros cannot achieve the same initializer usability, because any backing storage variables added must be initialized directly instead of supporting initialization through computed properties. For example, the proposed [`@Observable` macro](https://github.com/apple/swift-evolution/blob/main/proposals/0395-observability.md) applies a property-wrapper-like transform that turns stored properties into computed properties backed by the observation APIs, but it provides no way to write an initializer using the original property names like the programmer expects:
+
+```swift
+@Observable
+struct Proposal {
+  var title: String
+  var text: String
+
+  init(title: String, text: String) {
+    self.title = title // error: 'self' used before all stored properties are initialized
+    self.text = text // error: 'self' used before all stored properties are initialized
+  } // error: Return from initializer without initializing all stored properties
+}
+```
+
+## Proposed solution
+
+This proposal adds _`init` accessors_ to opt computed properties on types into virtual initialization that subsumes initialization of a set of zero or more specified stored properties, which allows assigning to computed properties in the body of a type's initializer:
+
+```swift
+struct Angle {
+  var degrees: Double
+  var radians: Double {
+    init(initialValue) initializes(degrees) {
+      degrees = initialValue * 180 / .pi
+    }
+
+    get { degrees * .pi / 180 }
+    set { degrees = newValue * 180 / .pi }
+  }
+
+  init(degrees: Double) {
+    self.degrees = degrees // sets 'self.degrees' directly
+  }
+
+  init(radians: Double) {
+    self.radians = radians // calls init accessor with 'radians'
```

Comment:
Heh, yeah I meant to describe that `radians` is passed as the argument to the init accessor, but I can see how that's completely unclear. I can change the name of the parameter so that it's more obvious.